tools/site_cobble/git_revision.py

```diff
@@ -0,0 +1,471 @@
+# This is python port of https://github.com/passsy/git-revision/
+# with some modifications to our preferences and use-cases.
+# See LICENSE.git_revision.md in the root of the repository.
+# -------------------------------------------------------------
+#
+#   Copyright 2021 Oxide Computer Company
```

Comment:
Does this need to be 2022 now? ha

Reply:
Yeah, if you look at the _git() method that actually runs the subprocess there's something like that there too but it has double decorators since functools.cache can't cache non-hashable or immutable data types and we're passing in a list cause that's what subprocess.run() wants.  So we decorate the function to turn the list into a tuple which is hashable and imutable and then convert back to a list to add the "git" command.  Provides a noticeable a speedup since git only needs to run new commands.